protocol-models/src/main/resources/airbyte_protocol/protocol_methods.md

```diff
@@ -0,0 +1,28 @@
+# Protocol Methods
+
+The purpose of this doc is to define the interface on the docker containers that implement the Airbyte Protocol. It is purposefully lean, as we describe the protocol at length in [our docs](https://docs.airbyte.com/understanding-airbyte/airbyte-protocol). We just want to define the interface here, so that as it evolves, we can version the change as part of the protocol version.
+
+## Method Interfaces
+
+We describe these interfaces in pseudocode for clarity. Clarifications on the pseudocode semantics:
+* Any `Stream~ that is mentioned as input arg, is passed to the docker contained via STDIN. 
+* All other parameters are passed in as command line args (e.g. --config <path to config file>). 
```

Comment:
Maybe a controversial opinion - I think trying to represent both the stdin/stdout values and the method parameters in the same step is confusing.

Maybe stating that the method returns an I/O stream and define what that I/O accepts and returns as a secondary step?

Reply:
FWIW, I also think this is a problem in our other docs describing these methods